docs/spec_checklist.md

```diff
@@ -56,3 +56,16 @@ reviews:
          for [ops_stablehlo.mlir](https://github.com/openxla/stablehlo/blob/main/stablehlo/tests/ops_stablehlo.mlir).
       1. Follow guideline [#6](https://github.com/openxla/stablehlo/blob/main/docs/reference_checklist.md#after-implementing-the-op)
          for [infer_stablehlo.mlir](https://github.com/openxla/stablehlo/blob/main/stablehlo/tests/infer_stablehlo.mlir).
+  1. Evaluate the op for [side effects and
+     speculatability](https://mlir.llvm.org/docs/Rationale/SideEffectsAndSpeculation/).
+      1. If the op has no side effects and is always speculatable, give it the
+         `Pure` trait. This is rare, as most ops allow dynamic shapes, which may
```

Comment:
Can we unify the AIs for no side effects, something like:

```
.., give it `Pure` and `NoMemoryEffect` traits. 
```

and drop the last line.

Reply:
`Pure` is `[AlwaysSpeculatable, NoMemoryEffect]`.